### PR TITLE
Fix licensing for Spring framework overrides

### PR DESCRIPTION
**Pull request by anguss00**
Remove the need for spock assert class importing

### CODE REVIEW

File: grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy [hunk 1]
```diff
@@ -1,7 +1,10 @@
 /*
  * Copyright 2015 IMOS
  *
- * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ * The AODN/IMOS Portal is licensed under the Apache License, Version 2.0 (the "License")
```
Comment (danfruehauf) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: The portal is licensed under GPLv3, only this file (and the other) are under the Apache license.

Comment (anguss00) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: Ok, I'll fix

Comment (anguss00) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: So we still need to say that the Portal is distributed under GPL here? And then just say that this file is specifically licensed under apache?

Comment (danfruehauf) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: I think say the portal is under GPLv3, but this file is under the Apache license + add header to Apache license.

Comment (danfruehauf) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: By all means, I'm not a lawyer. It's just that if it was my code here in the file, this is what I would have expected.

**Comment by anguss00** [comment]
changes made @danfruehauf

**Comment by danfruehauf** [comment]
That's good.

**Comment by anguss00** [comment]
:+1: thanks!

**Comment by danfruehauf** [comment]
I gonna test it first in the VM...

**Comment by anguss00** [comment]
@danfruehauf I have a solution, but I'm not sure how good it is...

File: grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy [hunk 2]
```diff
@@ -19,7 +29,7 @@ final class HtmlCharacterEntityReferences {
      * @since 4.1.2
      */
     public String convertToReference(char character, String encoding) {
-        if (encoding.startsWith("UTF")){
+        if (encoding.startsWith("UTF-")){
```
Comment (danfruehauf) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: Why this change?

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 3]
```diff
@@ -117,4 +116,9 @@ class LayerController {
     def _isXmlContent(contentType) {
         return contentType.find(/(text|application)\/xml/)
     }
+
+    def _contentTypeCharset(contentType) {
+        def stringComponents = contentType.split("=")
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: This is a tad bit too naive.

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 4]
```diff
@@ -117,4 +116,9 @@ class LayerController {
...
+        return stringComponents[1]
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: And this will return you a bunch of stuff you might not want. So generally speaking you should parse it properly or not parse it at all.

In the case of `Content-Type: application/xml;charset=UTF-8`, this will actually work fine, however think about
`Content-Type: application/xml;charset=UTF-8;value=booga`. Then in your case you gonna return `UTF-8;value` which is undesirable.

File: grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy (hunk 2 above)
Comment (anguss00) [inline grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy]: The return type from getEncoding() from an input stream was UTF8, while the value contained in the header is "UTF-8" (which is a better match for the internal groovy definition of the charset). I've taken the opportunity to tighten this conditional

**Comment by anguss00** [comment]
Changes made @danfruehauf... not sure why the commit name is still fixup after I used ``git rebase -i HEAD~2 `` and modified the commit label....

**Comment by anguss00** [comment]
Got rid of it, read the instructions @anguss :-p

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 5]
```diff
@@ -117,4 +116,14 @@ class LayerController {
     def _isXmlContent(contentType) {
         return contentType.find(/(text|application)\/xml/)
     }
+
+    def _extractCharsetType(contentType) {
+        def returnedCharset
+        contentType.split(';').each { token ->
+            if (token.contains('charset')) {
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Go with:
```
if (token.contains('charset=')) {
```

Be as strict as possible here, especially when we're doing "non-standard" parsing.

**Comment by danfruehauf** [comment]
It generally looks good. We'll just have to test it.

**Comment by anguss00** [comment]
Made that change @danfruehauf

**Comment by danfruehauf** [comment]
Let me test it quickly. I should have everything in place...

**Comment by danfruehauf** [comment]
Still doesn't work :(

File: grails-app/controllers/au/org/emii/portal/LayerController.groovy [hunk 6]
```diff
@@ -32,14 +32,13 @@ class LayerController {
...
                 def con = new URL(_getMetadataUrl(params.uuid)).openConnection()
-                def reader = new InputStreamReader(con.content)
                 def metadataText = con.content.text
 
                 if (_isXmlContent(con.contentType)) {
 
                     def xml = new XmlSlurper().parseText(metadataText)
                     //TODO: Validate schema before proceeding
-                    def abstractText = HtmlUtils.htmlEscape(xml.identificationInfo.MD_DataIdentification.abstract.CharacterString.text(), reader.getEncoding())
+                    def abstractText = HtmlUtils.htmlEscape(xml.identificationInfo.MD_DataIdentification.abstract.CharacterString.text(), _extractCharsetType(con.contentType))
```
Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: If I hard code the second parameter here to `UTF-8` and have `LC_ALL=C` - still breaks. Not sure what's the deal...

Comment (anguss00) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Wow... that's pretty depressing :-( not sure how we progress here...

Comment (danfruehauf) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: I'm not sure what's the deal. We'll have a look tomorrow perhaps?

Comment (anguss00) [inline grails-app/controllers/au/org/emii/portal/LayerController.groovy]: Sounds like a good plan, I'll mull it over in the meantime

**Comment by danfruehauf** [comment]
See #1767
